govcd/solution_addon_instance.go

```diff
@@ -0,0 +1,445 @@
+/*
+* Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"maps"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
```

Comment:
I see we have two occurrences of this import and add here the third. Would it be possible to part without this lib? Any 3rd party lib adds a toll on SRP.

CC: @dataclouder @adambarreiro

Reply:
Maybe we could simplify logging (I believe it's only used there, for tracing purposes).

Reply:
Simplified logging and used builtin `%#v` instead of additional package. Removed other occurrences and this dependency is no longer here.

Reply:
we have several occurrence of using `pretty.Formatter` in this repo. In some cases also in the main code, not only in testing.